casper/src/test/scala/coop/rchain/casper/MultiParentCasperRestartSpec.scala

```diff
@@ -0,0 +1,85 @@
+package coop.rchain.casper
+
+import java.nio.file.{Files, Path}
+
+import cats.effect._
+import coop.rchain.casper.helper.HashSetCasperTestNode
+import coop.rchain.casper.helper.HashSetCasperTestNode.{Effect, _}
+import coop.rchain.casper.scalatestcontrib._
+import coop.rchain.casper.util.{ConstructDeploy, ProtoUtil}
+import coop.rchain.crypto.signatures.Secp256k1
+import coop.rchain.p2p.EffectsTestInstances.LogicalTime
+import coop.rchain.shared.PathOps.RichPath
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Inspectors, Matchers}
+
+class MultiParentCasperRestartSpec extends FlatSpec with Matchers with Inspectors {
+
+  import MultiParentCasperTestUtil._
+
+  implicit val timeEff: LogicalTime[Effect] = new LogicalTime[Effect]
+
+  private val (validatorKeys, validatorPks) = (1 to 4).map(_ => Secp256k1.newKeyPair).unzip
+  private val genesis = buildGenesis(
+    buildGenesisParameters(4, createBonds(validatorPks))
+  )
+
+  "MultiParentCasper" should "load block data after restart" in effectTest {
+    coop.rchain.casper.helper.BlockDagStorageTestFixture.createDirectories[Effect].use {
+      case (blockStoreDir, blockDagDir) =>
+        Resource
+          .make[Effect, Path](
+            Sync[Effect].delay {
+              Files.createTempDirectory(s"casper-restart-spec-storage-")
+            }
+          )(dir => Sync[Effect].delay { dir.recursivelyDelete() })
+          .use { storageDir =>
+            for {
+              blockHash1 <- HashSetCasperTestNode
```

Comment:
This would be more readable if you named this computation

Reply:
Are you suggesting extracting this to a method with a meaningful name?

Reply:
yes. An inner val or def with a descriptive name would be nice